### ISSUE TITLE
`redirect` not honoring `basePath`

### ISSUE THREAD

**Issue by aryobenholzner** [vercel/next.js#53713]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.1.0
    Relevant Packages:
      next: 13.4.14-canary.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router, Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/gifted-forest-8kjr9w?file=%2Fnext.config.js%3A9%2C1

### To Reproduce

- add a `basePath` to `next.config.js` (e.g. `/base-path`)
- add a  `next/navigation` `redirect` to another page to a server page (e.g. `redirect('/sub-path')`)

### Describe the Bug

the URL changed to a path without including the `basePath`, resulting in Error 404

### Expected Behavior

like with `Link` from `next/link`, the base path should be prepended

**Comment by ajaykrathod** [comment]
Can you please assign me this issue? I want to look into this.

**Comment by aryobenholzner** [comment]
duplicated issue #54163 

fixed in #54277
